benchmark/parse_output.py

```diff
@@ -141,14 +142,109 @@ def parse_hgtector(input_f):
+
+    Notes
+    -----
+    genbank_fp is the intermediate GenBank file generated by reformat_input.py,
+    in which multiple sequences are concantenated, instead of the original
+    GenBank file.
+
+    Returns
+    -------
+    output: string
+        gene names (protein_ids) separated by newline
+    """
+    genes = {}
+    gb = Sequence.read(genbank_fp, format='genbank')
+    for feature in gb.interval_metadata.features:
+        if feature['type_'] == 'CDS':
+            if 'protein_id' in feature:
+                protein_id = feature['protein_id'].replace('\"', '')
+                if protein_id not in genes:
+                    loc = gb.interval_metadata.features[feature]
+                    start = loc[0][0] + 1
+                    end = loc[0][1]
+                    genes[protein_id] = (start, end)
+    genes_in_gi = {}
+    for line in input_f:
+        l = line.strip().split()
+        if len(l) < 2:
+            continue
+        (start, end) = (int(l[0]), int(l[1]))
+        for (gene, l) in genes.items():
+            if (l[0] >= start and l[1] <= end):
+                if gene not in genes_in_gi:
+                    genes_in_gi[gene] = 1
+    return '\n'.join(sorted(genes_in_gi))
+
+
+def parse_genemark(input_f, genbank_fp):
+    """ Extract atypical genes identified by GeneMark
+
+    Parameters
+    ----------
+    input_f: string
+        file descriptor for GeneMark output gene list (*.lst)
+    genbank_fp: string
+        file path to genome in GenBank format
+
+    Notes
+    -----
+    genbank_fp is the intermediate GenBank file generated by reformat_input.py,
+    in which multiple sequences are concantenated, instead of the original
+    GenBank file.
+
+    Returns
+    -------
+    output: string
+        gene names (protein_ids) separated by newline
+    """
+    genes = {}
+    gb = Sequence.read(genbank_fp, format='genbank')
+    for feature in gb.interval_metadata.features:
+        if feature['type_'] == 'CDS':
+            if 'protein_id' in feature:
+                protein_id = feature['protein_id'].replace('\"', '')
+                if protein_id not in genes:
+                    strand = '-' if feature['rc_'] else '+'
+                    loc = gb.interval_metadata.features[feature]
+                    start = loc[0][0] + 1
+                    end = loc[0][1]
+                    genes[protein_id] = (start, end, strand)
+    atypical_genes = []
+    reading = False
+    for line in input_f:
+        l = line.strip().split()
+        if len(l) == 2 and l == ['#', 'Length']:
+            reading = True
+        elif reading and len(l) == 6 and l[5] == '2':
+            (start, end, strand) = (int(l[2].lstrip('<>')),
+                                    int(l[3].lstrip('<>')),
+                                    l[1])
+            for (gene, l) in genes.items():
+                if l[0] == start and l[1] == end and l[2] == strand:
+                    atypical_genes.append(gene)
```

Comment:
you're right, do you mind adding a comment above that line to specify that? Thanks :)